### PR TITLE
added hw-5-1-solution

### PR DESCRIPTION
**Pull request by AlinaPodoliaka**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,64 @@
...
+
+    public Engine getEngine() {
+        return engine;
+    }
+
+    public void setEngine(Engine engine) {
+        this.engine = engine;
+    }
+
+    public Car clone() throws CloneNotSupportedException {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java]: this method should be overriden

File: src/main/java/core/basesyntax/Car.java [hunk 2]
```diff
@@ -1,5 +1,64 @@
...
+        Car newCar = (Car) super.clone();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java]: try to solve without super.clone()

File: src/main/java/core/basesyntax/Engine.java [hunk 3]
```diff
@@ -1,5 +1,62 @@
...
+    public String getCooling() {
+        return cooling;
+    }
+
+    public void setCooling(String cooling) {
+        this.cooling = cooling;
+    }
+
+    public Engine clone() throws CloneNotSupportedException {
+        return (Engine) super.clone();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Engine.java]: same

File: src/main/java/core/basesyntax/Car.java (hunk 1 above)
Comment (AlinaPodoliaka) [reply src/main/java/core/basesyntax/Car.java]: You mean I must write "@overriding"? But in Idea already shown icon informing about overriding...

Comment (Tarabarnik) [reply src/main/java/core/basesyntax/Car.java]: https://beginnersbook.com/2014/07/override-annotation-in-java/

File: src/main/java/core/basesyntax/Car.java [hunk 4]
```diff
@@ -1,5 +1,63 @@
...
+        return engine;
+    }
+
+    public void setEngine(Engine engine) {
+        this.engine = engine;
+    }
+
+    @Override
+    public Car clone() throws CloneNotSupportedException {
+        Car newCar = new Car(this.model, this.color, this.maxSpeed, this.mileage, this.engine);
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java]:
```suggestion
        Car newCar = new Car(model, color, maxSpeed, mileage, engine.clone());
```

File: src/main/java/core/basesyntax/Engine.java [hunk 5]
```diff
@@ -1,5 +1,65 @@
...
+        return cooling;
+    }
+
+    public void setCooling(String cooling) {
+        this.cooling = cooling;
+    }
+
+    @Override
+    public Engine clone() throws CloneNotSupportedException {
+        Engine newEngine = new Engine(this.engineVolume, this.enginePower, this.numberOfCylinders,
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Engine.java]:
```suggestion
        Engine newEngine = new Engine(engineVolume, enginePower, numberOfCylinders,
```

File: src/main/java/core/basesyntax/Engine.java [hunk 6]
```diff
@@ -1,5 +1,65 @@
...
+                this.fuelType, this.cooling);
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Engine.java]:
```suggestion
                fuelType, cooling);
```

File: src/main/java/core/basesyntax/Car.java (hunk 1 above)
Comment (AlinaPodoliaka) [reply src/main/java/core/basesyntax/Car.java]: Pavel, thank you for article. I understood, why I must write theese annotation.

Comment (AlinaPodoliaka) [reply src/main/java/core/basesyntax/Car.java]: Please, review my file once more.

**Pull request closed without merging**